source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -710,6 +710,16 @@ Why is waitQueueTimeoutMS optional for some drivers?
 
 We are anticipating eventually introducing a single client-side timeout mechanism, making us hesitant to introduce another granular timeout control. Therefore, if a driver/language already has an idiomatic way to implement their timeouts, they should leverage that mechanism over implementing waitQueueTimeoutMS.
 
+Why is there a distinction between a Pool being "closed" and the Pool resources being free-d / garbage collected? (a.k.a. why does a Pool have a "half-closed" state?)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are a few motivating factors for this decision:
+
+#. When a pool is closed, there may be connections belonging to the pool that are still checked out. In many situations, it would be very dangerous for the pool to aggressively close these checked-out connections. Therefore, **when a pool is closed, it does not aggresively close checked-out connections**.
+#. Blocking the driver on waiting for all connections to be closed by a pool could lead to performance hits, and in some cases deadlocks. Therefore, **when the driver closes a pool, it does not block on all connections being destroyed** (an exception may be made for application shutdown).
```

Comment:
I guess it would not be an exact deadlock. My thinking was a situation where:

a. Connection **A** is checked out of Pool **B**
b. Pool close is started
c. Lock b/c:
  - **B** is waiting for **A** to be checked in before finishing the close
  - The thread that holds **A** is waiting for **B*8 to close before checking in the connection

I'm not sure exactly how that is possible, but it seems like something that *could* happen.

That said, it seems unlikely, so I'd be okay rewording it to just `Blocking the driver on waiting for all connections to be closed by a pool could lead to performance hits`

Reply:
I still don't get it.  What is `**B*8` referring to?

Reply:
Updated to correct my markdown.